cantools/database/can/c_source.py

```diff
@@ -566,6 +567,14 @@
     {type_name} {name}{length};\
 '''
 
+INIT_SIGNAL_BODY_TEMPLATE_FMT = '''\
+    {type_name} tmp_{signal_name} = {signal_initial};
+    memcpy((void*)(&msg_p->{signal_name}),
+           (void*)&tmp_{signal_name},
+           sizeof({type_name}));
```

Comment:
why does a simple
```C
msg_p->{signal_name} = {signal_initial};
```
not work here?

Reply:
Yes, it works.
From my point of view, *memcpy()* is able to produce more efficient code

Reply:
this is certainly not true here: you first copy the data conventionally,  so you have two copy operations. also, -- in contrast to `memcpy` -- the compiler is aware of the exact properties of the type to be copied, so it  almost always produces code that is at least as fast...

Reply:
But you are correct, I cannot work around the first assignment.
I'll fix it